core/src/test/scala/unit/kafka/server/KafkaApisTest.scala

```diff
@@ -0,0 +1,95 @@
+import org.apache.kafka.common.protocol.{SecurityProtocol, ApiKeys, ProtoUtils}
+import org.apache.kafka.common.requests.{RequestSend, RequestHeader, HeartbeatRequest}
+import org.junit.Test
+import org.junit.Assert.{assertNull, assertNotNull}
+
+/**
+ * Unit test for KafkaApis
+ */
+class KafkaApisTest {
+  //TODO: The API version test should use KafkaServerTestHarness with actual clients after we have protocol version control on clients.
+  @Test(expected = classOf[UnsupportedVersionException])
+  def testValidateRequestVersionForScalaRequests() {
+    val fetchRequest = new FetchRequest((ProtoUtils.latestVersion(ApiKeys.FETCH.id) + 1).toShort, 0, "test", -1, 100, 1,
+      Map.empty[TopicAndPartition, PartitionFetchInfo])
+    val buffer = RequestOrResponseSend.serialize(fetchRequest)
+    buffer.rewind()
+    val request = RequestChannel.Request(0, "foo", new Session(null, "bar"), buffer, 0L, SecurityProtocol.PLAINTEXT)
+    assertNull(request.header)
+    assertNull(request.body)
+    KafkaApis.validateRequestVersion(request, Option(request.requestObj.asInstanceOf[FetchRequest].versionId))
+  }
+
+  @Test(expected = classOf[UnsupportedVersionException])
+  def testValidateRequestVersionForJavaRequests() {
+    val heartBeatRequest = new HeartbeatRequest("foo", 0, "bar")
+    val requestHeader = new RequestHeader(ApiKeys.HEARTBEAT.id, "foo", 0)
+    val requestSend = new RequestSend("foo", requestHeader, heartBeatRequest.toStruct)
+    val buffer = ByteBuffer.allocate(requestSend.size().toInt)
+    val simpleGatheringBytesChannel = new SimpleGatheringBytesChannel(buffer)
+    requestSend.writeTo(simpleGatheringBytesChannel)
+    // Skip first 4 bytes for the size delimited format and change the API version.
+    buffer.putShort(6, (ProtoUtils.latestVersion(ApiKeys.HEARTBEAT.id) + 1).toShort)
+    buffer.position(4)
+    val request = RequestChannel.Request(0, "foo", new Session(null, "bar"), buffer.slice(), 0L, SecurityProtocol.PLAINTEXT)
+    assertNotNull(request.header)
+    assertNull(request.body)
+    KafkaApis.validateRequestVersion(request, None)
+  }
+
+  // This class is written because the NetworkSend can only write to a GatheringByteChannel. This class should be
+  // removed after the unit test start to use KafkaTestHarness.
+  private class SimpleGatheringBytesChannel(buffer: ByteBuffer) extends GatheringByteChannel {
+    override def write(buffers: Array[ByteBuffer], offset: Int, length: Int) : Long = {
+      0L
+    }
+
+    override def write(buffers: Array[ByteBuffer]): Long = {
+      val positionBeforeWrite = buffer.position()
+      buffers foreach buffer.put
+      buffer.position() - positionBeforeWrite
+    }
+
+    override def write(byteBuffer: ByteBuffer): Int = {
+      val positionBeforeWrite = buffer.position()
+      buffer.put(byteBuffer)
+      buffer.position() - positionBeforeWrite
+    }
+
+    override def isOpen: Boolean = {
+      true
+    }
+
+    override def close() {
+
+    }
```

Comment:
I think it's better to just close the brace in the same line as the opening brace. Whitespace is good to structure things, but we should not waste it where it doesn't help. :)